src/common/components/addressForm/addressForm.component.js

```diff
@@ -23,19 +23,24 @@ class AddressFormController {
       .subscribe((data) => {
         this.countries = data;
         if(this.address){
-          this.refreshRegions(this.address.country);
+          this.refreshRegions(this.address.country, true);
         }
       });
   }
 
-  refreshRegions(country){
+  refreshRegions(country, initial = false){
     country = find(this.countries, {name: country});
     if(!country){ return; }
```

Comment:
Would this stop your clearing code from running? I'm not really sure when it wouldn't find a country.